test/Orleans.Transactions.Tests/Runners/OrchestrationsTransactionsTestRunner.cs

```diff
@@ -108,18 +113,40 @@ protected OrchestrationsTransactionsTestRunner(IGrainFactory grainFactory, ITest
             await coordinator.MultiGrainAdd(grains, delta);
 
             await grains[0].Get();
-            await CheckReport(grainId, 2, 0, 2);
+            await CheckReport(grainId, 2, 0);
         }
 
-        private async Task CheckReport(Guid grainId, int perpareCount, int abortCount, int commitCount)
+        private async Task CheckReport(Guid grainId, int perpareCount, int abortCount)
+        {
+            var endTime = DateTime.UtcNow + this.waitTime;
+            while (DateTime.UtcNow < endTime)
+            {
+                bool passed = await CheckReport(grainId, perpareCount, abortCount, false);
+                if (passed) return;
+                await Task.Delay(this.waitTime.Milliseconds / 10);
+            }
+            await CheckReport(grainId, perpareCount, abortCount, true);
+        }
+
+        private async Task<bool> CheckReport(Guid grainId, int perpareCount, int abortCount, bool assert)
         {
             var resultGrain = this.grainFactory.GetGrain<ITransactionOrchestrationResultGrain>(grainId);
 
             TransactionOrchestrationResult results = await resultGrain.GetResults();
 
-            Assert.Equal(perpareCount, results.Prepared.Count);
-            Assert.Equal(abortCount, results.Aborted.Count);
-            Assert.Equal(commitCount, results.Committed.Count);
+            if(assert)
+            {
+                Assert.Equal(perpareCount, results.Prepared.Count);
+                Assert.Equal(abortCount, results.Aborted.Count);
+                Assert.Equal(results.Prepared.Max(), results.Committed.Aggregate((long)0, (t1, t2) => Math.Max(t1, t2)));
```

Comment:
Not entirely sure I follow this logic, as I'm not familiar with this structure we are using. But it kind of looks like you should be using `Sum` instead of `Max` if intermediate commits are batched, or am I misunderstanding the assert?

Reply:
Shouldn't the check be "max prepared = max committed" in which case, `.Max()` on both collections?

Reply:
The prepare is called for every transaction, so the list should have all the transaction ID's in it.  The commit is called for at least the highest transaction.  So the max of the prepared should match the max of the committed transaction id's.  I use an aggregate function rather than a link max, because link max will throw an exception when there are 0 items, and aggregate will not.